### DISCUSSION TITLE
[Feature Request] Z-Lift after "Abort Print Job"

### DISCUSSION THREAD

**Discussion by endpointmanager** [mriscoc/Ender3V2S1#709] [Q&A]
**This is a feature request.**

Scenario: Print something using OctoPrint, abort print job using the printer's display during the very first layer.

Sometimes, I have to abort the print job during the very first layer using the printers display e.g. if I detect bad warping or wrong Z-offset after a nozzle/surface-change. Unfotunately, the firmware just moves Y axis to it's max-position, and does not lift Z in advance, which might cause the nozzle crash into the warped object.

Or even worse, If your Z-offset is (for some reason) wrong, aborting using the display makes it even worse.

I would love to see the possibility to lift the nozzle (Z-axis) before it moves the bed (Y-axis) when someone aborts the printjob.

PS: Thanks for developing this awesome firmware!!!

**Comment by mriscoc** [comment]
Hi, if you are printing from Octoprint, is it who controls the movements, check in [Octoprint settings](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)  the script `After print job is cancelled` and add the Z lift there.

**Reply by endpointmanager** [reply]
Hi, I already know the possibility to abort printjobs using Octoprint and add custom g-code there. However, the computer I remote-control Ocoprint is not in the same room, therefore I abort printjobs using the printer's display, and if I do so I'm pretty sure the firmware is responsible to control movements during abortion. Same happens without using Octoprint at all. (Print something from the sdcard, and then abort it)

What I'm looking for, is an option to lift Z when the firmware is responsible for movements immediately after it aborted the print job.

**Reply by mriscoc** [reply]
If the G-code file is on the SD card and Octoprint started it, the firmware controls the abort procedure. If the G-code file is in Octoprint, the printer just sends an abort signal to Octoprint when you press cancel on the display unit and Octoprint takes control.
